config/runtime/kunit.jinja2

```diff
@@ -5,20 +5,83 @@
 
 {%- block python_imports %}
 {{ super() }}
+import json
 import subprocess
 {%- endblock %}
 
 {%- block python_globals %}
 {{ super() }}
 REVISION = {{ revision }}
+STATUS_MAP = {
+    'PASS': 'pass',
+    'FAIL': 'fail',
+    'SKIP': None,
+}
 {% endblock %}
 
-{%- block python_body %}
-def main(argv):
-    src_path = argv[1]
+{% block python_job -%}
+class Job(BaseJob):
+    def _parse_results(self, data):
+        child_nodes = []
+        node = {
+            'name': data['name'],
+            'child_nodes': child_nodes,
+        }
+        for test_case in data.get('test_cases', []):
+            child_nodes.append({
+                'name': test_case['name'],
+                'result': STATUS_MAP[test_case['status']],
+            })
+        for test_group in data.get('sub_groups', []):
+            child_nodes.append(self._parse_results(test_group))
+        return node
 
-    print("Running KUnit...")
-    cmd = f"cd {src_path} && ./tools/testing/kunit/kunit.py run --json=kunit.json --jobs=$(nproc)"
-    res = subprocess.call(cmd, shell=True)
-    return res == 0
+    def _run(self, src_path):
+        kunit_json = 'kunit.json'
+
+        print("Running KUnit...")
+        cmd = f"\
+cd {src_path} && \
+./tools/testing/kunit/kunit.py run --json={kunit_json} --jobs=$(nproc)"
+        if subprocess.call(cmd, shell=True):
```

Comment:
@gctucker why is this a specific case to use `call()` instead of `run()`?

Reply:
Force of habit I guess.  Now I see that `subprocess.run()` is the new and more generic method to use so I can convert it to use that instead.

https://docs.python.org/3/library/subprocess.html#subprocess.call